llvm_util/llvm2alive.cpp

```diff
@@ -870,6 +870,14 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
         return error(i);
       RETURN_IDENTIFIER(make_unique<Free>(*b, false));
     }
+    case llvm::Intrinsic::trap:
+    {
+      FnAttrs attrs;
+      attrs.set(FnAttrs::NoReturn);
+      attrs.set(FnAttrs::NoRead);
```

Comment:
Would trap also be NoWrite?
Since trap is noreturn, its output memory won't be used anyway, but I wasn't not sure whether we can say  it is really NoWrite. :/

Reply:
I think NoWrite should be fine. We still have the UB part to ensure that trap is called.
BTW, can you add a test to ensure trap can't be refined with unreach, like:
trap
unreachable
=>
unreachable

Reply:
I'm less sure about noread. We need to double check the semantics. Is this ok?
```
store 1, foo
trap
=>
store 2, foo
trap
```

(please add this test as well)

Reply:
LangRef says llvm.trap may be lowered to a target-specific instruction, so it seems noread is a bit unsafe. :/ I removed it.
I checked that LLVM unit tests that contain `llvm.trap` passed after this change.